packages/core/saved-objects/core-saved-objects-api-server/src/apis/find.ts

```diff
@@ -66,11 +66,23 @@ export interface SavedObjectsFindOptions {
    * Use `hasReferenceOperator` to specify the operator to use when searching for multiple references.
    */
   hasReference?: SavedObjectsFindOptionsReference | SavedObjectsFindOptionsReference[];
+
   /**
    * The operator to use when searching by multiple references using the `hasReference` option. Defaults to `OR`
    */
   hasReferenceOperator?: 'AND' | 'OR';
 
+  /**
+   * Search for documents *not* having a reference to the specified objects.
+   * Use `hasNoReferenceOperator` to specify the operator to use when searching for multiple references.
+   */
+  hasNoReference?: SavedObjectsFindOptionsReference | SavedObjectsFindOptionsReference[];
+
+  /**
+   * The operator to use when searching by multiple references using the `hasNoReference` option. Defaults to `OR`
+   */
+  hasNoReferenceOperator?: 'AND' | 'OR';
```

Comment:
Note for later: 

https://github.com/elastic/kibana/blob/34c228b0c941308e01298e31223e7af0a5a2cee9/packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/search_dsl/query_params.ts#L131

Should be extracted to `@kbn/core-saved-objects-api-server`